### ISSUE TITLE
Crashing on android 11

### ISSUE THREAD

**Issue by siddharth-kt** [Vydia/react-native-background-upload#288]
My app is crashing on android 11 only and i got this report is firebase crashlytics :

Fatal Exception: java.lang.RuntimeException: Unable to create service net.gotev.uploadservice.UploadService: java.lang.IllegalArgumentException: You have to set namespace to your app package name (context.packageName) in your Application subclass
       at android.app.ActivityThread.handleCreateService(ActivityThread.java:4537)
       at android.app.ActivityThread.access$1700(ActivityThread.java:265)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2171)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:257)
       at android.app.ActivityThread.main(ActivityThread.java:8185)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:626)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1015)

Caused by java.lang.IllegalArgumentException: You have to set namespace to your app package name (context.packageName) in your Application subclass
       at net.gotev.uploadservice.UploadServiceConfig.getNamespace(UploadServiceConfig.kt:74)
       at net.gotev.uploadservice.UploadServiceConfig.getBroadcastNotificationAction(UploadServiceConfig.kt:234)
       at net.gotev.uploadservice.UploadServiceConfig.getBroadcastNotificationActionIntentFilter(UploadServiceConfig.kt:248)
       at net.gotev.uploadservice.observer.request.NotificationActionsObserver.register(NotificationActionsObserver.kt:31)
       at net.gotev.uploadservice.UploadService.onCreate(UploadService.kt:194)
       at android.app.ActivityThread.handleCreateService(ActivityThread.java:4517)
       at android.app.ActivityThread.access$1700(ActivityThread.java:265)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2171)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:257)
       at android.app.ActivityThread.main(ActivityThread.java:8185)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:626)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1015)

Kindly tell me the solution!!

**Comment by sergeymild** [comment]
I have the same error

**Comment by quangphungntq** [comment]
![Screen Shot 2021-10-16 at 6 23 45 PM](https://user-images.githubusercontent.com/42831325/137585536-c359b9cd-802a-4bf2-af1f-b69c342386a1.png)
I have the same error
Can anyone help us?

**Comment by swizes** [comment]
Edit 

**node_modules/react-native-background-upload/android/build.gradle b/node_modules/react-native-background-upload/android/build.gradle**

```
-    implementation 'net.gotev:uploadservice-okhttp:4.6.0'
+    implementation 'net.gotev:uploadservice-okhttp:4.7.0'
```

Then apply the patch via **npx patch-package react-native-background-upload**

More info at
https://github.com/ds300/patch-package#readme

**Comment by hackaprende** [comment]
Error is still happening to me
Version 6.6.0
RN Version 0.66.4

Tried to add 
packages.add(new UploaderReactPackage());

on MainApplication.java getPackages() as it says in the README but Then I get another error:
`[java.lang.IllegalStateException: Native module RNC_AsyncSQLiteDBStorage tried to override AsyncStorageModule. Check the getPackages() method in MainApplication.java, it might be that module is being created twice. If this was your intention, set canOverrideExistingModule=true. This error may also be present if the package is present only once in getPackages() but is also automatically added later during build time by autolinking. Try removing the existing entry and rebuild.]`

**Comment by Hassaan68** [comment]
> Error is still happening to me Version 6.6.0 RN Version 0.66.4
> 
> Tried to add packages.add(new UploaderReactPackage());
> 
> on MainApplication.java getPackages() as it says in the README but Then I get another error: `[java.lang.IllegalStateException: Native module RNC_AsyncSQLiteDBStorage tried to override AsyncStorageModule. Check the getPackages() method in MainApplication.java, it might be that module is being created twice. If this was your intention, set canOverrideExistingModule=true. This error may also be present if the package is present only once in getPackages() but is also automatically added later during build time by autolinking. Try removing the existing entry and rebuild.]`

Hi, Have you got any solution for this ?

**Comment by r0xsh** [comment]
> Error is still happening to me Version 6.6.0 RN Version 0.66.4
> 
> Tried to add packages.add(new UploaderReactPackage());
> 
> on MainApplication.java getPackages() as it says in the README but Then I get another error: `[java.lang.IllegalStateException: Native module RNC_AsyncSQLiteDBStorage tried to override AsyncStorageModule. Check the getPackages() method in MainApplication.java, it might be that module is being created twice. If this was your intention, set canOverrideExistingModule=true. This error may also be present if the package is present only once in getPackages() but is also automatically added later during build time by autolinking. Try removing the existing entry and rebuild.]`

This is another issue, you have to delete `packages.add(new UploaderReactPackage());` since now libraries are [autolinked](https://github.com/react-native-community/cli/blob/main/docs/autolinking.md). We should update the README

**Comment by noumantahir** [comment]
still happening at large, any proposed fix so far..

<img width="971" alt="Screenshot 2024-06-12 at 17 04 56" src="https://github.com/Vydia/react-native-background-upload/assets/6298342/7c5083c6-3cf1-44e1-a195-d74c2c5d9c8f">

**Comment by juanlaborde1895** [comment]
I'm facing the same issue right now, @noumantahir have you been able to find a solution?

**Comment by alfarodarwaynejay** [comment]
I'm facing the same issue too, I tried

```
-    implementation 'net.gotev:uploadservice-okhttp:4.6.0'
+    implementation 'net.gotev:uploadservice-okhttp:4.7.0'
```

but it did not work

**Comment by Franco-Julian** [comment]
@juanlaborde1895 @alfarodarwaynejay have you find the way to solved this?

**Comment by alfarodarwaynejay** [comment]
> @juanlaborde1895 @alfarodarwaynejay have you find the way to solved this?

No, I am planning on creating my own upload using WebView.

Something like [this]( https://uploadcare.com/blog/file-uploading-in-react-native/)

**Comment by scremona-navico** [comment]
Add this to your build.gradle:
android {
    // Conditional for compatibility with AGP <4.2.
    if (project.android.hasProperty("namespace")) {
        namespace "com.vydia.RNUploader"
    } else {
        // print
        println "DEPRECATION WARNING: The `namespace` property is not available in your version of AGP. Please upgrade to AGP 4.2+."
    }
...

then you'll probably need this fix too:
https://github.com/Vydia/react-native-background-upload/issues/343

**Comment by sobanarshad85** [comment]
try this patch file It'll fix the issue

 # Reason for the 1st patch is to fix compatibility issue with Android 14
 # see https://github.com/Vydia/react-native-background-upload/issues/343
 # see https://github.com/Vydia/react-native-background-upload/pull/347

diff --git a/node_modules/react-native-background-upload/android/build.gradle b/node_modules/react-native-background-upload/android/build.gradle
index fa6963f..7a3c908 100755
--- a/node_modules/react-native-background-upload/android/build.gradle
+++ b/node_modules/react-native-background-upload/android/build.gradle
@@ -1,9 +1,9 @@
 buildscript {
     ext {
-        kotlinVersion = '1.6.0'
-        buildToolsVersion = '29.0.2'
-        compileSdkVersion = 29
-        targetSdkVersion = 29
+        kotlinVersion = '1.8.21'
+        buildToolsVersion = '34.0.8'
+        compileSdkVersion = 34
+        targetSdkVersion = 34
         minSdkVersion = 18
     }
     ext.detoxKotlinVersion = ext.kotlinVersion
@@ -20,7 +20,7 @@ buildscript {

 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
+apply plugin: 'kotlin-parcelize'

 def DEFAULT_COMPILE_SDK_VERSION = 28
 def DEFAULT_BUILD_TOOLS_VERSION = "28.0.3"
@@ -64,7 +64,7 @@ dependencies {

     implementation "org.jetbrains.kotlin:$_kotlinStdlib:$_kotlinVersion"

-    implementation 'net.gotev:uploadservice-okhttp:4.7.0'
+    implementation 'net.gotev:uploadservice-okhttp:4.9.2'

     implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.0.0'
 }
